src/controllers/products.js

```diff
@@ -16,30 +16,28 @@ exports.createProduct = async (req, res, next) => {
   const product = await Product.create(req.body);
   return res.status(201).json({ success: true, data: product });
 };
-exports.updateProduct = async (req, res) => {
+exports.updateProduct = async (req, res, next) => {
   // check for validation errors first
   const validationErrors = validationResult(req);
   if (!validationErrors.isEmpty()) {
     return res
       .status(400)
       .json({ success: false, errors: validationErrors.array() });
   }
+  // eslint-disable-next-line no-cond-assign, no-undef
+  const myproduct = await Product.findById(req.params.id);
```

Comment:
It's better to name it something more meaningful like, `product` or at least `myProduct`.